src/vs/vscode.proposed.d.ts

```diff
@@ -877,6 +877,17 @@ declare module 'vscode' {
 
 	//#endregion
 
+	//#region Terminal name change event https://github.com/microsoft/vscode/issues/114898
+
+	export namespace window {
+		/**
+		 * An event which fires when the name of the terminal changes.
+		 */
+		export const onDidChangeTerminalName: Event<Terminal>;
+	}
```

Comment:
I'll look at this more thoroughly next week, but I was expecting `Pseudoterminal.onDidChangeName` to allow "Extension Terminals" to control their name, instead of this general event for all terminals.

Reply:
Ah sorry, I misunderstood the initial requirements. I've added a new commit but I may have bitten off more than I can chew on this one